vpr/src/place/place.cpp

```diff
@@ -891,7 +891,11 @@ void try_place(const t_placer_opts& placer_opts,
     } while (update_annealing_state(&state, success_rat, costs, placer_opts, annealing_sched));
     /* Outer loop of the simmulated annealing ends */
 
+#ifdef ENABLE_ANALYTIC_PLACE
+// guard quench label, otherwise compiler complains about unused label
 quench:
```

Comment:
Didn't realize there was now a goto in place.cpp.  This would be good to restructure. It could be restructured by:

#ENABLE_ANALYTIC_PLACE wouldn't guard the command line option, but would print a message saying you have to compile with that #define to use it.  Now you can test on placer_opts.enable_analytic_placer anywhere you want.

Then restructure the do_while for the annealer into a while loop controlled like 
exit_anneal = !placer_opts.enable_analytic_placer  // Skip the anneal with analytic placer, otherwise start the anneal
while (!exit_anneal) {

 .... lots of stuff
  exit+anneal = update_annealing_state (...)

Not sure if it is best to land the PR first or restructure first -- the restructuring is pretty local.

Reply:
I had a conversation with Bill and he told me to use goto for now and he will take care of it when he refactors place.cpp later. I think he changed the original while loop to do_while so perhaps he has some preferences. @Bill-hbrhbr.

Reply:
OK, thanks.  Will merge when kokoro finishes then.  Thinking about this more, we don't need to change the command line option guards to get rid of the goto.  Just need:
```
exit_anneal = false;
#ifdef ENABLE_ANALYTIC_PLACE;
  if (placer_opts.enable_analytic_placer)
     exit_anneal = true;
# endif

// change to while loop as above.
```

Reply:
Hi Vaughn, the CI tests have passed. Bill confirmed with me that he will remove the goto when he refactors place.cpp. I'm currently running more tests and will update the result in PR description probably by tomorrow. It there's nothing else to change, perhaps you could merge the branch.